### PR TITLE
Start iteration 5

### PR DESCRIPTION
**Pull request by brisag**
Hi all!
We started iteration 5.  I think its a good time to break and make sure everything else is refactored for all our other files.

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -260,4 +260,55 @@ def item_revenue(item_id)
...
+
+  def item_count_by_merchants(customer_id)
+    item_count_by_merchant = group_invoices_by_merchant_id(customer_id)
+    item_count_by_merchant.each do |merchant, invoices|
+      item_count_by_merchant[merchant] = sum_of_item_quantity_for_invoice(invoices)
+    end
+  end
+
+  def top_merchant_for_customer(customer_id)
+    max_items = item_count_by_merchants(customer_id).max_by do |key, value|
```
Comment (brisag) [inline lib/sales_analyst.rb]: done.

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -260,4 +260,55 @@ def item_revenue(item_id)
...
+  def group_invoices_by_merchant_id(customer_id)
+    @engine.find_all_invoices_by_customer_id(customer_id).group_by do |invoice|
+      invoice.merchant_id
+    end
+  end
+
+  def item_count_by_merchants(customer_id)
+    item_count_by_merchant = group_invoices_by_merchant_id(customer_id)
+    item_count_by_merchant.each do |merchant, invoices|
+      item_count_by_merchant[merchant] = sum_of_item_quantity_for_invoice(invoices)
```
Comment (brisag) [inline lib/sales_analyst.rb]: made it shorter by indenting to new line

File: test/sales_analyst_test.rb [hunk 3]
```diff
@@ -294,4 +294,84 @@ def test_it_can_find_best_item_for_merchant
...
+    assert_instance_of Array, expected_array
+    assert_instance_of Invoice, expected_array.first
+    assert_instance_of Hash, analyst.group_invoices_by_merchant_id(7)
+  end
+
+  def test_item_count_by_merchants
+    engine = SalesEngine.from_csv(@csv_data)
+    analyst = SalesAnalyst.new(engine)
+    expected = {
+                12334861 => 37,
```
[the remark this answers is not in the document]
Comment (brisag) [reply test/sales_analyst_test.rb]: fixed indent

**Review by Diana20920 — APPROVED** [review]
Less Hound comments do feel and look nice 💪🏼

**Review by DougWelchons — APPROVED** [review]
Looks good we can open up a new branch to start for refactoring to dial in what we've got.

**Pull request merged**